src/promptflow/promptflow/_sdk/_service/apis/flow.py

```diff
@@ -38,54 +42,103 @@
     "FlowUxInput",
     {
         "flow": fields.String(required=True, description="Path to flow directory."),
+        "ux_inputs": fields.Nested(dict_field, required=True, description="Flow ux inputs"),
     }
 )
 
+flow_path_parser = reqparse.RequestParser()
+flow_path_parser.add_argument('flow', type=str, required=True, location='args', help='Path to flow directory.')
+
 
 @api.route("/test")
 class FlowTest(Resource):
     @api.response(code=200, description="Flow test", model=dict_field)
     @api.doc(description="Flow test")
     @api.expect(flow_test_model)
     def post(self):
-        flow = api.payload["flow"]
+        args = flow_path_parser.parse_args()
+        flow = args.flow
+        flow = decrypt_flow_path(flow)
         inputs = api.payload.get("inputs", None)
         environment_variables = api.payload.get("environment_variables", None)
         variant = api.payload.get("variant", None)
         node = api.payload.get("node", None)
         experiment = api.payload.get("experiment", None)
         output_path = api.payload.get("output_path", None)
-        if Configuration.get_instance().is_internal_features_enabled() and experiment:
-            result = get_client_from_request().flows.test(
-                flow=flow,
-                inputs=inputs,
-                environment_variables=environment_variables,
-                variant=variant,
-                node=node,
-                experiment=experiment,
-                output_path=output_path,
-            )
-        else:
-            result = get_client_from_request().flows.test(
-                flow=flow,
-                inputs=inputs,
-                environment_variables=environment_variables,
-                variant=variant,
-                node=node,
-                allow_generator_output=False,
-                stream_output=False,
-                dump_test_result=True,
-                output_path=output_path,
-            )
-        return result
+        remove_dir = False
+
+        if output_path is None:
+            filename = str(uuid.uuid4())
+            if os.path.isdir(flow):
+                output_path = Path(flow) / PROMPT_FLOW_DIR_NAME / filename
+            else:
+                output_path = Path(os.path.dirname(flow)) / PROMPT_FLOW_DIR_NAME / filename
+            os.makedirs(output_path, exist_ok=True)
+            remove_dir = True
+        output_path = Path(output_path).resolve()
+        try:
+            if Configuration.get_instance().is_internal_features_enabled() and experiment:
+                result = get_client_from_request().flows.test(
```

Comment:
Could we refactor here, some logic here seems better suit to put in flow_operations. Ideal implementation might be update pass some in memory storage can call executor.

Reply:
refactored